### ISSUE TITLE
can_sam.c fails to compile when CONFIG_NETWORKING is set

### ISSUE THREAD

**Issue by kendag** [zephyrproject-rtos/zephyr#64691]
**Describe the bug**
Attempting to build an application with both CONFIG_CAN and CONFIG_NETWORKING enabled produces a compilation error.

**To Reproduce**
`west build -b sam_e70_xplained tests/drivers/can/api/ -- -DCONFIG_NETWORKING=y
`
**Expected behavior**
The test harness should build without error.

**Impact**
Currently unable to add CAN functionality to my application which also uses networking.

**Logs and console output**
```
drivers/can/can_sam.c:169:44: error: section of 'can_sam_mram_0' conflicts with previous declaration
  169 |         CAN_MCAN_DT_INST_MRAM_DEFINE(inst, can_sam_mram_##inst);        \
      |                                            ^~~~~~~~~~~~~
```

**Environment (please complete the following information):**
 - OS: Linux 6.3.12-200.fc38.x86_64
 - Toolchain: Zephyr SDK 0.16.3
 - Version: zephyr-v3.5.0

**Additional context**

**Comment by henrikbrixandersen** [comment]
@nandojve FYI

**Comment by henrikbrixandersen** [comment]
This is a clash between `__no_init` and `__no_cache` resulting in different sections with `CONFIG_NOCACHE_MEMORY=y`. Looking into a solution.

**Comment by henrikbrixandersen** [comment]
@kendag Please test the proposed fix in #65169 and report back if it solves your issue. It compiles here and should work, but I do not have the hardware for testing.
